modules/edu.gemini.giapi.client/src/main/scala/edu/gemini/giapi/client/package.scala

```diff
@@ -0,0 +1,107 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package edu.gemini.giapi
+
+import cats.implicits._
+import cats.effect._
+import edu.gemini.aspen.giapi.util.jms.status.StatusGetter
+import edu.gemini.jms.activemq.provider.ActiveMQJmsProvider
+import fs2.async
+
+package object client {
+
+  /**
+    * Allowed types according to GIAPI
+    */
+  implicit val strItemGetter: ItemGetter[String]    = new ItemGetter[String] {}
+  implicit val doubleItemGetter: ItemGetter[Double] = new ItemGetter[Double] {}
+  implicit val intItemGetter: ItemGetter[Int]       = new ItemGetter[Int]    {}
+  implicit val floatItemGetter: ItemGetter[Float]   = new ItemGetter[Float]  {}
+}
+
+package client {
+
+  /**
+    * Typeclass to present as evidence when calling `Giapi.get`
+    */
+  sealed trait ItemGetter[A]
+
+  /**
+    * Represents a connection to a GIAPi based instrument
+    *
+    * @tparam F Effect type
+    */
+  trait GiapiConnection[F[_]] {
+    def connect: F[Giapi[F]]
+  }
+
+  /**
+    * Algebra to interact with a GIAPI instrument
+    *
+    * @tparam F Effect Type
+    */
+  trait Giapi[F[_]] {
+
+    /**
+      * Returns a value for the status item. If not found or there is an error, e.g. on types the exception is returned
+      */
+    def get[A: ItemGetter](statusItem: String): F[Option[A]]
+
+    /**
+      * Close the connection
+      */
+    def close: F[Unit]
```

Comment:
I think this should be in GiapiConnection but if I put it there, I can't get Stream.bracket to access close